### ISSUE TITLE
Incorrect starting position and weird animation inside NavigationView

### ISSUE THREAD

**Issue by Dara-To** [Wouter125/BottomSheet#11]
I am using relative positioning inside a NavigationView. The issue about the wrong position and enter animation makes this library unusable. If there is no NavigationView, there is no issue. But I do need it.

```
enum BottomSheetPosition: CGFloat, CaseIterable {
    case top = 0.83
    case middle = 0.385
}

@State var bottomSheetPosition: BottomSheetPosition = .middle

BottomSheetView(
                position: $bottomSheetPosition,
                header: {
                },
                content: {
                    ForecastView(bottomSheetTranslationProrated: bottomSheetTranslationProrated)
                    
                }
            )
            .onBottomSheetDrag { translation in
                bottomSheetTranslation = translation

                withAnimation(.easeInOut) {
                    if bottomSheetPosition == BottomSheetPosition.top {
                        hasDragged = true
                    } else {
                        hasDragged = false
                    }
                }
            }
```

https://user-images.githubusercontent.com/33760969/170066308-c2393fbb-04ed-47d0-a618-1f2695e4746f.mov

**Comment by Wouter125** [comment]
Thank you for your input and the bug that you found! Could you provide me with some more code of how you wrap it in a NavigationView? I just setup a small NavigationView inside the example and it works. Not seeing any weird animation glitches.

https://user-images.githubusercontent.com/6507123/170131942-7e75b783-e15e-4763-ac44-c47f45157af8.mov

**Comment by Wouter125** [comment]
I just included this example in the latest release that I did. Please check it out to see if it helps you figure out where the problem is at!

**Comment by Dara-To** [comment]
Thanks for taking the time to test it with the navigation view. With your insights, I tested more and found out the culprit was my background image. To be more specific, it's the `resizable` modifier. I am not sure why since the bottom sheet and the background image and color are layered in a ZStack. Also, that's how one should format the background image and even add `scaledToFill()`. Of course, the animation flickering is still absent without navigation view. And adding geometry reader and setting the frame on the background image didn't help either. 

```
ZStack {
                Color.background
                    .ignoresSafeArea()
                
                Image("Background")
                    .resizable()
                    .ignoresSafeArea()
                
                Image("House")
                    .frame(maxHeight: .infinity, alignment: .top)
                    .padding(.top, 257)
                
                content

                BottomSheetView(
                    position: $bottomSheetPosition,
                    header: {
                    },
                    content: {
                        ForecastView()
                    }
                )
                .onBottomSheetDrag { translation in
                    bottomSheetTranslation = translation

                    withAnimation(.easeInOut) {
                        if bottomSheetPosition == BottomSheetPosition.top {
                            hasDragged = true
                        } else {
                            hasDragged = false
                        }
                    }
                }
            }
        }
```

**Comment by Wouter125** [comment]
I just rebuild my example with your background and set it to resizable. Unfortunately I'm still not able to reproduce the problem. Since it's animating at the very beginning I feel like it has to do with something that runs within an `withAnimation` block. Could you share a minimal example (on a Github repo) that reproduces the issue? Happy to take a look and see where the issue is coming from!

https://user-images.githubusercontent.com/6507123/170821570-44d1546a-80ed-47cc-b22e-ee383bf8c249.mov

**Comment by Dara-To** [comment]
I just restarted the project until the bottom sheet step, committed to Github to a new repo and invited you. Thanks for looking into it.

**Comment by Wouter125** [comment]
Thanks for the sample project. I was able to fix your issue in #13. If you update the package to the latest version it should work. If you run into any other issues let me know!

**Comment by Dara-To** [comment]
My God, you're so fast! It's fixed of me, thanks! I'm glad we got to resolve it before I started recording for my course next week. Thanks a bunch! What was the exact cause?

**Comment by Wouter125** [comment]
The BottomSheet was placed in a NavigationView but the frame of the BottomSheet was set to zero. So the NavigationView picked it up as a change and animated the view. I added a small little check to see if the view was already laid out and if not I disabled the animations. After the view is laid out I toggle the boolean and animations will be used from that moment on. The code can be found here:

https://github.com/Wouter125/BottomSheet/pull/13/commits/deee138526707dc6d7f1e8b9a6657a56590d4553
